Penteract/Assets/Scripts/PlayerController.cpp

```diff
@@ -209,6 +219,30 @@ void PlayerController::SwitchCharacter() {
 			switchInProgress = true;
 			playSwitchParticles = false;
 		}
+		if (switchCollisionedGO.size() > 0) {
+			for (GameObject* enemy : switchCollisionedGO) {
+				AIMeleeGrunt* meleeScript = GET_SCRIPT(enemy, AIMeleeGrunt);
+				RangedAI* rangedScript = GET_SCRIPT(enemy, RangedAI);
+				if (rangedScript || meleeScript) {
+					if (meleeScript) {
+						meleeScript->EnableBlastPushBack();
+						if (switchFirstHit) {
+							meleeScript->gruntCharacter.GetHit(switchDamage);
+							meleeScript->PlayHit();
+						}
+					}
+					else if (rangedScript) {
+						rangedScript->EnableBlastPushBack();
+						if (switchFirstHit) {
+							rangedScript->rangerGruntCharacter.GetHit(switchDamage);
+							rangedScript->PlayHit();
+						}
+					}
+				}
+			}
```

Comment:
I think this code should go inside the "if (doVisualSwitch)" statement so the pushback is synchronized with the blast switch effect / model swapping and not triggering at the beginning as it does right now. I think this way you can avoid dealing with the switchFirstHit as the code inside this if statement just runs once per switch.
 
This should be a decision made by the game design team